client_src/js/Tour.mjs

```diff
@@ -117,39 +120,56 @@ class Tour {
 			},
 			buttons: this.get_buttons()
 		});
-		
-		
+
+
 		this.tour.addStep("complete", {
 			text: "Tour complete!\nIf you need any additional assistance, let us know :-)",
-			buttons: [{ text: "Done", action: this.tour.next }]
+			buttons: this.get_buttons(true, false, false)
 		});
 	}
-	
+
 	run_once() {
 		if(window.localStorage.getItem("completed_tour") === null)
 			this.run();
-		
-		window.localStorage.setItem("completed_tour", (new Date()).toISOString());
+
+		//window.localStorage.setItem("completed_tour", (new Date()).toISOString()); -- removed due to an accidental refresh possiblity
 	}
-	
+
 	run() {
 		if(typeof this.tour == "undefined")
 			this.create_tour();
 		this.tour.start();
 	}
-	
-	get_buttons(no_continue = false, no_prev = false) {
+
+	completed_tour()
+	{
+		window.localStorage.setItem("completed_tour", (new Date()).toISOString());
+	}
+
+	get_buttons(no_continue = false, no_prev = false, no_end = true) {
 		let next = { text: "Next", action: this.tour.next },
 			prev = { text: "Previous", action: this.tour.back },
-			exit = { text: "Exit", action: this.tour.cancel };
-		
+			exit = { text: "Exit", action: () =>
+			{
```

Comment:
Small thing, but perhaps the opening brace `{` of the arrow function could be on the same line as the `=>` please? Same goes for below too.